components/tikv_util/src/sys/cgroup.rs

```diff
@@ -1,520 +1,236 @@
-        assert_eq!(-1, cgroup.read_num("cpu.cfs_quota_us").unwrap());
-
-        // Read invalid number
-        let path = tmp_dir.path().join("cpu.cfs_period_us");
-        let mut f3 = File::create(path).unwrap();
-        f3.write_all(b"abc").unwrap();
-        f3.sync_all().unwrap();
-        assert!(cgroup.read_num("cpu.cfs_period_us").is_err());
-
-        // Read number with \n from file `memory.max_usage`.
-        let path = tmp_dir.path().join("memory.max_usage");
-        let mut f1 = File::create(path).unwrap();
-        f1.write_all(b"123\n").unwrap();
-        f1.sync_all().unwrap();
-        assert_eq!(123, cgroup.read_num("memory.max_usage").unwrap());
+    fn test_parse_cpu_cores() {
+        let mut cpusets = Vec::new();
+        cpusets.extend(parse_cpu_cores("1-2,5-8,10,12,4"));
+        cpusets.sort();
+        assert_eq!(cpusets, vec![1, 2, 4, 5, 6, 7, 8, 10, 12]);
     }
 
     #[test]
-    fn test_read_cpuset() {
-        use std::io::{Seek, SeekFrom};
-
-        let tmp_dir = TempDir::new().unwrap();
-        let cgroup = CGroup::new(tmp_dir.path().to_str().unwrap().to_string());
-
-        // Read number from file `memory.limits_in_bytes`
-        let path = tmp_dir.path().join(CPUSET_CPUS);
-        let mut f = File::create(&path).unwrap();
-        f.sync_all().unwrap();
-        assert_eq!(0, cgroup.read_cpuset().unwrap().len());
-
-        f.set_len(0).unwrap();
-        f.seek(SeekFrom::Start(0)).unwrap();
-        f.write_all(b"0\n").unwrap();
-        f.sync_all().unwrap();
-        assert_eq!(1, cgroup.read_cpuset().unwrap().len());
-
-        f.set_len(0).unwrap();
-        f.seek(SeekFrom::Start(0)).unwrap();
-        f.write_all(b"1,2,3\n").unwrap();
-        f.sync_all().unwrap();
-        assert_eq!(3, cgroup.read_cpuset().unwrap().len());
-
-        f.set_len(0).unwrap();
-        f.seek(SeekFrom::Start(0)).unwrap();
-        f.write_all(b"4-6\n").unwrap();
-        f.sync_all().unwrap();
-        assert_eq!(3, cgroup.read_cpuset().unwrap().len());
-
-        f.set_len(0).unwrap();
-        f.seek(SeekFrom::Start(0)).unwrap();
-        f.write_all(b"7,8-9,10,11-12\n").unwrap();
-        f.sync_all().unwrap();
-        assert_eq!(6, cgroup.read_cpuset().unwrap().len());
+    fn test_parse_cpu_quota_v2() {
+        let contents = vec!["max 100000", "10000 100000", "1000000 100000"];
+        let expects = vec![None, Some(0.1), Some(10.0)];
+        for (content, expect) in contents.into_iter().zip(expects) {
+            let limit = parse_cpu_quota_v2(content);
+            assert_eq!(limit, expect);
+        }
     }
 
     #[test]
-    fn test_cgroup_sys() {
-        let temp_dir = TempDir::new().unwrap();
-        let path1 = temp_dir.path().join("cgroup");
-        let mut f1 = File::create(path1.clone()).unwrap();
-        f1.write_all(b"4:memory:/kubepods/burstable/poda2ebe2cd-64c7-11ea-8799-eeeeeeeeeeee/a026c487f1168b7f5442444ac8e35161dfcde87c175ef27d9a806270e267a575\n").unwrap();
```

Comment:
I think this can be reserved. It's important that the code can parse files in real life.

Reply:
Now it's added into `test_parse_proc_cgroup_v1`.